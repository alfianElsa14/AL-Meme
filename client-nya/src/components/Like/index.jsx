import React from 'react';
import { useDispatch } from 'react-redux';
import { FormattedMessage } from 'react-intl';
import { useParams } from 'react-router-dom';

import { addLike, removeLike } from '@pages/MemeDetail/actions';

import ThumbUpOffAltIcon from '@mui/icons-material/ThumbUpOffAlt';
import ThumbUpAltIcon from '@mui/icons-material/ThumbUpAlt';
import classes from './style.module.scss';

function Like({ likes, userId }) {
  const dispatch = useDispatch();
  const { id } = useParams();
  const userLiked = likes?.dataLike?.some((like) => like?.userId === userId);

  const handleLikeSubmit = () => {
    if (userLiked) {
      dispatch(removeLike(id));
    } else {
      dispatch(addLike(id));
    }
  };

  return (
    <div
      data-testid="Like"
      className={classes.like}>
      {userLiked ? (
        <ThumbUpAltIcon onClick={handleLikeSubmit} className={classes.iconLike} />
      ) : (
        <ThumbUpOffAltIcon onClick={handleLikeSubmit} className={classes.iconLike}
          data-testid="iconLike"
        />
      )}
      <p>
        {likes?.like} <FormattedMessage id='app_likes' />
      </p>
    </div>
  );
}

export default Like;
