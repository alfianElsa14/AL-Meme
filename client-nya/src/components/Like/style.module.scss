@import '@styles/base';

.like {
    display: flex;
    align-items: center;
    gap: 10px;

    @include apply-at-md-size {
        padding-top: 30px;
    }

    .iconLike {
        color: $color-aqua;
        cursor: pointer;
    }
}