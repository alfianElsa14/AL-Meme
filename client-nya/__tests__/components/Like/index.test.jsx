import Like from "@components/Like";
import classes from '@components/Like/style.module.scss';
import { render } from "@utils/testHelper";

const mockPropsLike = [
    {
        userId,
        memeId,
        createdAt: new Date(),
        updatedAt: new Date(),
        User: {
            id: 1,
            username: 'mikail',
            email: 'mikail@gmail.com',
            imageUrl: 'image.jpg',
            role: 'basic',
            verified: true,
            verifyToken: null,
            createdAt: new Date(),
            updatedAt: new Date()
        }
    },
]

let wrapper;
let userId = mockPropsLike[0].User.id
let memeId = 1

beforeEach(() => {
    wrapper = render(<Like likes={mockPropsLike} userId={userId} />)
})

describe("Like", () => {
    test('Should render correctly', () => {
        const { getByTestId } = wrapper;
        const commentComponent = getByTestId('Like');
        expect(commentComponent).toHaveClass(classes.like)
        expect(commentComponent).toBeInTheDocument();
    })
    
    test('Should render icon like correctly', () => {
        const { getByTestId } = wrapper;
        const commentComponent = getByTestId('iconLike');
        expect(commentComponent).toHaveClass(classes.iconLike)
        expect(commentComponent).toBeInTheDocument();
    })

    test('Should match with snapshot', () => {
        expect(wrapper).toMatchSnapshot();
    })
})