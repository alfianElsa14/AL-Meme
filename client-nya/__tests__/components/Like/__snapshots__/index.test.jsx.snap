// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Like Should match with snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="like"
        data-testid="Like"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium iconLike css-i4bv87-MuiSvgIcon-root"
          data-testid="iconLike"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="m13.11 5.72-.57 2.89c-.12.59.04 1.2.42 1.66.38.46.94.73 1.54.73H20v1.08L17.43 18H9.34c-.18 0-.34-.16-.34-.34V9.82l4.11-4.1M14 2 7.59 8.41C7.21 8.79 7 9.3 7 9.83v7.83C7 18.95 8.05 20 9.34 20h8.1c.71 0 1.36-.37 1.72-.97l2.67-6.15c.11-.25.17-.52.17-.8V11c0-1.1-.9-2-2-2h-5.5l.92-4.65c.05-.22.02-.46-.08-.66-.23-.45-.52-.86-.88-1.22L14 2zM4 9H2v11h2c.55 0 1-.45 1-1v-9c0-.55-.45-1-1-1z"
          />
        </svg>
        <p>
           
          Suka
        </p>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="like"
      data-testid="Like"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium iconLike css-i4bv87-MuiSvgIcon-root"
        data-testid="iconLike"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="m13.11 5.72-.57 2.89c-.12.59.04 1.2.42 1.66.38.46.94.73 1.54.73H20v1.08L17.43 18H9.34c-.18 0-.34-.16-.34-.34V9.82l4.11-4.1M14 2 7.59 8.41C7.21 8.79 7 9.3 7 9.83v7.83C7 18.95 8.05 20 9.34 20h8.1c.71 0 1.36-.37 1.72-.97l2.67-6.15c.11-.25.17-.52.17-.8V11c0-1.1-.9-2-2-2h-5.5l.92-4.65c.05-.22.02-.46-.08-.66-.23-.45-.52-.86-.88-1.22L14 2zM4 9H2v11h2c.55 0 1-.45 1-1v-9c0-.55-.45-1-1-1z"
        />
      </svg>
      <p>
         
        Suka
      </p>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
